src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -16,21 +16,21 @@ namespace System
     [Serializable]
     [NonVersionable] // This only applies to field layout
     [TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    public partial struct Guid : IFormattable, IComparable, IComparable<Guid>, IEquatable<Guid>, ISpanFormattable
+    public readonly partial struct Guid : IFormattable, IComparable, IComparable<Guid>, IEquatable<Guid>, ISpanFormattable
     {
         public static readonly Guid Empty = default;
 
-        private int _a;   // Do not rename (binary serialization)
-        private short _b; // Do not rename (binary serialization)
-        private short _c; // Do not rename (binary serialization)
-        private byte _d;  // Do not rename (binary serialization)
-        private byte _e;  // Do not rename (binary serialization)
-        private byte _f;  // Do not rename (binary serialization)
-        private byte _g;  // Do not rename (binary serialization)
-        private byte _h;  // Do not rename (binary serialization)
-        private byte _i;  // Do not rename (binary serialization)
-        private byte _j;  // Do not rename (binary serialization)
-        private byte _k;  // Do not rename (binary serialization)
+        private readonly int _a;   // Do not rename (binary serialization)
+        private readonly short _b; // Do not rename (binary serialization)
+        private readonly short _c; // Do not rename (binary serialization)
+        private readonly byte _d;  // Do not rename (binary serialization)
+        private readonly byte _e;  // Do not rename (binary serialization)
+        private readonly byte _f;  // Do not rename (binary serialization)
+        private readonly byte _g;  // Do not rename (binary serialization)
+        private readonly byte _h;  // Do not rename (binary serialization)
+        private readonly byte _i;  // Do not rename (binary serialization)
+        private readonly byte _j;  // Do not rename (binary serialization)
+        private readonly byte _k;  // Do not rename (binary serialization)
```

Comment:
Is marking the fields as readonly going to break binary serialization?

Reply:
Just did a simple experiment of creating a struct with a non-readonly field, serialising it to a file, marking the field and the struct as readonly and deserialising it; It appears to work fine without exceptions. Though, I haven't really used binary serialisation so I could be wrong.

Reply:
Please verify the reverse as well since that's also supported (although, if ->readonly works, surely <-readonly does too)

Reply:
The opposite seems to be true as well.